### PR TITLE
Binance websocket: Order Detail Execution Report

### PR DESCRIPTION
**Pull request by yangrq1018**
# PR Description

- Set more fields of order.Detail
- "ExecutionReport" only provides "trade" status (partially filled), so check that executed amount == total amount before change status to filled

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/order/order_types.go [hunk 1]
```diff
@@ -133,6 +133,7 @@ type Detail struct {
 	ExecutedAmount       float64
 	RemainingAmount      float64
 	Cost                 float64
+	CostAsset            string
```
Comment (shazbert) [inline exchanges/order/order_types.go:136]: Can we change this type from string to currency.Code? This is just for differentiation purposes.

**Review by shazbert — CHANGES_REQUESTED** [review]
@yangrq1018 Just one minor issue, thanks for opening this PR.

File: exchanges/binance/binance_websocket.go [hunk 2]
```diff
@@ -258,6 +258,8 @@ func (b *Binance) wsHandleData(respRaw []byte) error {
 					Date:            data.Data.OrderCreationTime,
 					Pair:            p,
 					ClientOrderID:   clientOrderID,
+					Cost:            data.Data.Commission,
+					CostAsset:       currency.NewCode(data.Data.CommissionAsset),
```
Comment (gloriousCode) [inline exchanges/binance/binance_websocket.go:262]: This change has broken `TestWsOrderExecutionReport`. Please check whether `data.Data.CommissionAsset != ""` first before creating the currency code.
Could you please also update the test? The json has `"n":"0", "N": null,`. If you could update it to have values and compare them that would be much appreciated.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thank you for your PR yangrq1018! 🎉 Just one thing from me

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for quickly making those changes!
# 🎉

**Review by shazbert — APPROVED** [review]
tACK! Thanks.

**Review by thrasher- — APPROVED** [review]
tACK, thanks @yangrq1018 !
